static/js/search/search_input.tsx

```diff
@@ -0,0 +1,285 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Input box with autocomplete for Search Page.
+ */
+
+import _ from "lodash";
+import React, { useEffect, useRef, useState } from "react";
+
+import { NamedNode, NamedPlace } from "../shared/types";
+import { getPlaceDcids, getPlaceNames } from "../utils/place_utils";
+import {
+  getHighlightedJSX,
+  getStatVarSearchResults,
+} from "../utils/search_utils";
+
+interface SearchInputPropType {
+  query: string;
+}
+const NUM_SV_RESULTS = 5;
+const DELAY_MS = 200;
+const REDIRECT_PREFIX = "/search2?";
+
+export function SearchInput(props: SearchInputPropType): JSX.Element {
+  const delayTimer = useRef(null);
+  const placeAutocompleteService = useRef(null);
+  const latestQuery = useRef(null);
+  const [hoveredIdx, setHoveredIdx] = useState(0);
+  const [results, setResults] = useState({ placeResults: [], svResults: [] });
+  const [inputText, setInputText] = useState(props.query);
+  const [showResults, setShowResults] = useState(false);
+
+  useEffect(() => {
+    if (google.maps) {
+      placeAutocompleteService.current = new google.maps.places.AutocompleteService();
+    }
+  }, []);
+
+  const matches = inputText.split(" ");
+  return (
+    <>
+      <div
+        className="search-input-section"
+        onBlur={(event) => {
+          if (!event.currentTarget.contains(event.relatedTarget as Node)) {
+            setShowResults(false);
+          }
+        }}
+        onKeyDown={(event) => handleKeydownEvent(event)}
+      >
+        <div className="search-input-wrapper">
+          <form
+            onSubmit={(event) => {
+              event.preventDefault();
+              redirectAction(inputText, "");
+            }}
+            tabIndex={-1}
+          >
+            <input
+              className="search-input-box"
+              type="text"
+              value={inputText}
+              onChange={(event) => onInputChanged(event.target.value)}
+              placeholder="Search Data Commons"
```

Comment:
Maybe the placeholder can be long, eg "Search for places, variables and more on Data Commons"

Reply:
done